src/main/java/com/coreos/jetcd/ClientBuilder.java

```diff
@@ -114,20 +115,53 @@ public ClientBuilder setNameResolverFactory(
    *
    * @return nameResolverFactory
    */
-  public AbstractEtcdNameResolverFactory getNameResolverFactory() {
+  public NameResolver.Factory getNameResolverFactory() {
     return nameResolverFactory;
   }
 
+  public ManagedChannelBuilder<?> getChannelBuilder() {
+    return channelBuilder;
+  }
+
+  public ClientBuilder setChannelBuilder(ManagedChannelBuilder<?> channelBuilder) {
+    this.channelBuilder = channelBuilder;
+    return this;
+  }
+
+  public boolean isLazyInitialization() {
+    return lazyInitialization;
+  }
+
+  /**
+   * Define if the client has to initialize connectivity and authentication on client constructor
+   * or delay it to the first call to a client. Default is false.
+   *
+   * @param lazyInitialization true if the client has to lazily perform connectivity/authentication.
+   * @return this builder
+   */
+  public ClientBuilder setLazyInitialization(boolean lazyInitialization) {
+    this.lazyInitialization = lazyInitialization;
+    return this;
+  }
+
   /**
    * build a new Client.
    *
    * @return Client instance.
    * @throws ConnectException As network reason, wrong address
    * @throws AuthFailedException This may be caused as wrong username or password
    */
-  public Client build() throws ConnectException, AuthFailedException {
+  public Client build() {
     checkState(!endpoints.isEmpty() || nameResolverFactory != null,
         "please configure etcd server endpoints or nameResolverFactory before build.");
-    return new ClientImpl(null, this);
+    return new ClientImpl(this);
+  }
+
+  public ClientBuilder copy() {
+    try {
+      return (ClientBuilder)super.clone();
+    } catch (CloneNotSupportedException e) {
+      throw new RuntimeException(e);
```

Comment:
`throw EtcdExceptionFactory.newEtcdException(e)`?
I would like all runtime exceptions throw as type of `EtcdException`.